### PR TITLE
cranelift: add i64.{ishl,ushr,ashr} libcalls

### PR DESCRIPTION
**Pull request by whitequark**
These libcalls are useful for 32-bit platforms.

On x86_32 in particular, commit 4ec16fa0 added support for legalizing 64-bit shifts through SIMD operations. However, that legalization requires SIMD to be enabled and SSE 4.1 to be supported, which is not acceptable as a hard requirement.

(Actually, when implementing that commit I didn't yet realize that these shifts ought to be legalized via libcalls rather than expansions most of the time, but the expansions are just fine.)

With this commit it is finally possible to use `cargo run --target i686-unknown-linux-gnu run foo.wasm` without having to add `--enable-simd`.

### CODE REVIEW

File: cranelift/codegen/src/legalizer/mod.rs [hunk 1]
```diff
@@ -35,7 +35,7 @@ mod table;
 use self::call::expand_call;
 use self::globalvalue::expand_global_value;
 use self::heap::expand_heap_addr;
-use self::libcall::expand_as_libcall;
+pub use self::libcall::expand_as_libcall;
```
Comment (bjorn3) [inline cranelift/codegen/src/legalizer/mod.rs:38]:
```suggestion
pub(crate) use self::libcall::expand_as_libcall;
```

File: cranelift/codegen/src/isa/x86/enc_tables.rs [hunk 2]
```diff
@@ -1376,6 +1378,8 @@ fn convert_ushr(
     let mut pos = FuncCursor::new(func).at_inst(inst);
     pos.use_srcloc(inst);
 
+    let isa_flags = isa_settings::Flags::new(&isa.flags(), isa_settings::builder());
```
Comment (abrown) [inline cranelift/codegen/src/isa/x86/enc_tables.rs:1381]: Just a drive-by comment: I'm not too sure this will work. To get the ISA flags I had to add some extra machinery (see https://github.com/bytecodealliance/wasmtime/pull/1759/commits/849d71bdb7bbcc3c6f6975dfb2224e75185dab06)

Comment (abrown) [reply cranelift/codegen/src/isa/x86/enc_tables.rs:1381]: And it's use: https://github.com/bytecodealliance/wasmtime/pull/1759/commits/6ab817eae953cfb4fd76b62d94bf89408e289b94#diff-f5aa2e18f8d875720c00f30b86cb69d4R1523

Comment (whitequark) [reply cranelift/codegen/src/isa/x86/enc_tables.rs:1381]: Yeah, you're right, it doesn't work. Which PR is that machinery added in?

**Comment by whitequark** [comment]
Looks like this PR depends on #1759.

File: cranelift/codegen/src/isa/x86/enc_tables.rs (hunk 2 above)
Comment (abrown) [reply cranelift/codegen/src/isa/x86/enc_tables.rs:1381]: #1759, which should be ready to go; @bnjbvr, here's another user for the ISA-specific flag machinery I added.

Comment (abrown) [reply cranelift/codegen/src/isa/x86/enc_tables.rs:1381]: (Oh, sorry, @whitequark, I see you already figured that out below!)

**Comment by whitequark** [comment]
@abrown I rebased the PR on master. I think some checks are failing because of unrelated issues.

**Comment by abrown** [comment]
That's annoying; do you mind opening an issue for it? Looks like something to do with lightbeam and staticvec from [this](https://github.com/bytecodealliance/wasmtime/pull/1787/checks?check_run_id=736850359#step:14:16) and [this](https://github.com/bytecodealliance/wasmtime/pull/1787/checks?check_run_id=736850244#step:4:217)

**Comment by whitequark** [comment]
> That's annoying; do you mind opening an issue for it?

I'm confused, isn't master supposed to be always green?

**Comment by iximeow** [comment]
Yes, but some tasks are nightly, with a recent nightly change invalidating what was otherwise fine. https://github.com/bytecodealliance/wasmtime/pull/1818 appears to fix this, try a rebase?

**Comment by whitequark** [comment]
Rebased

**Comment by whitequark** [comment]
Looks like it's green now!

**Comment by whitequark** [comment]
Thanks!
